### PR TITLE
Handle pre-release versions in Flutter version parsing and comparison

### PR DESCRIPTION
**Pull request by kenzieschmoll**
This PR also uses this new version parsing code to determine whether to show the legacy or new performance page.

### CODE REVIEW

File: packages/devtools_app/lib/src/version.dart [hunk 1]
```diff
@@ -17,18 +17,34 @@ class FlutterVersion extends SemanticVersion {
     @required this.engineRevision,
     @required this.dartSdkVersion,
   }) {
+    final splitOnDash = version.split('-');
```
Comment (jacob314) [inline packages/devtools_app/lib/src/version.dart:20]: comment what the format of a pre release version is expected to look like here.

File: packages/devtools_app/lib/src/version.dart [hunk 2]
```diff
@@ -17,18 +17,34 @@ class FlutterVersion extends SemanticVersion {
...
-        .split('.')
-        .map((part) => RegExp(r'\d+').stringMatch(part) ?? '0')
-        .toList();
+    final _versionParts = semVersion.split('.');
     major =
         _versionParts.isNotEmpty ? int.tryParse(_versionParts.first) ?? 0 : 0;
     minor = _versionParts.length > 1 ? int.tryParse(_versionParts[1]) ?? 0 : 0;
     patch = _versionParts.length > 2 ? int.tryParse(_versionParts[2]) ?? 0 : 0;
+
+    if (splitOnDash.length == 2) {
```
Comment (jacob314) [inline packages/devtools_app/lib/src/version.dart:34]: assert splitOnDash.length <=2

**Review by jacob314 — APPROVED** [review]
lgtm

File: packages/devtools_app/lib/src/version.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/version.dart:20]: done

File: packages/devtools_app/lib/src/version.dart (hunk 2 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/version.dart:33]: done

**Comment by kenzieschmoll** [comment]
Build is broken due to unrelated Inspector Golden failures. Will send a follow up PR to fix the bots.

**Pull request merged**